src/web/Settings.re

```diff
@@ -239,6 +265,71 @@ module Update = {
           ...settings,
           explainThis,
         };
+      | Assistant(u) =>
+        switch (u) {
+        | UpdateChatStatus => {
+            ...settings,
+            assistant: {
+              ...settings.assistant,
+              ongoing_chat: !settings.assistant.ongoing_chat,
+            },
+          }
+        | SwitchMode(mode) => {
+            ...settings,
+            assistant: {
+              ...settings.assistant,
+              mode,
+            },
+          }
+        | ToggleHistory => {
+            ...settings,
+            assistant: {
+              ...settings.assistant,
+              show_history: !settings.assistant.show_history,
+            },
+          }
+        | SetLLM(llm_id) =>
```

Comment:
we should find another way to do this... this module should really just be about a functional updating of the settings, we shouldn't be doing effects here. these should be actions in a separate assistant component

Reply:
this one is writing to disk though, and the other is issuing an API request. the other settings actions just update the model. this is not necessarily a dealbreaker but definitely a smell